cmd/namespace.go

```diff
@@ -38,3 +294,96 @@ func askToSelectNamespace(l log.Logger, conf *config.ClientConfig) (*config.Name
 		return namespace, nil
 	}
 }
+
+func getValidatedNamespaces(clientConfig *config.ClientConfig, selectedNamespaceNames []string) (validNames []string, invalidNames []string) {
+	if len(selectedNamespaceNames) == 0 {
+		return
+	}
+	for _, name := range selectedNamespaceNames {
+		if _, err := clientConfig.GetNamespaceByName(name); err != nil {
+			invalidNames = append(invalidNames, name)
+		} else {
+			validNames = append(validNames, name)
+		}
+	}
+	return
+}
+
+func registerAllNamespaces(l log.Logger, clientConfig *config.ClientConfig) error {
+	var selectedNamespaceNames []string
+	for _, namespace := range clientConfig.Namespaces {
+		selectedNamespaceNames = append(selectedNamespaceNames, namespace.Name)
+	}
+	return registerSelectedNamespaces(l, clientConfig, selectedNamespaceNames)
+}
+
+func registerSelectedNamespaces(l log.Logger, clientConfig *config.ClientConfig, selectedNamespaceNames []string) error {
```

Comment:
avoid registerAllNamespaces and have functions for registerNamespaces instead of registerSelectedNamespaces. Pass all the arguments needed for registering and avoid passing client config.

Reply:
it's addressed